lib/ansible/config/manager.py

```diff
@@ -116,7 +116,7 @@ def get_config_type(cfile):
 
     ftype = None
     if cfile is not None:
-        ext = to_text(os.path.splitext(cfile)[-1])
+        ext = os.path.splitext(cfile)[-1]
```

Comment:
What's the motivation for removing `to_text` here?

Reply:
The first commit of this pull-request reverts a2596f888700c7d97201c269ce6348830b46d6de.

Reply:
Yes, but why?

Reply:
Using Python 3 and a2596f8, `ansible-config -c ansible.cfg dump` remains unusable. a2596f8 fixes an exception but the configuration file is still not read: [`ConfigParser.read`](https://github.com/python/cpython/blob/master/Lib/configparser.py#L690) doesn't support bytes:
```
(Pdb) bt
  ansible/bin/ansible-config(118)<module>()
-> exit_code = cli.run()
  ansible/lib/ansible/cli/config.py(72)run()
-> self.config = ConfigManager(self.config_file)
  ansible/lib/ansible/config/manager.py(202)__init__()
-> self._parse_config_file()
  ansible/lib/ansible/config/manager.py(220)_parse_config_file()
-> self._parser.read(cfile)
> /usr/lib/python3.6/configparser.py(696)read()
-> with open(filename, encoding=encoding) as fp:
```
```
(Pdb) filenames
b'path/to/ansible.cfg'
(Pdb) [x for x in filenames]
[... 47, 97, 110, 115, 105, 98, 108, 101, 46, 99, 102, 103]
```
Avoiding `bytes` conversion fixes both problem, then the conversion added by a2596f8 becomes useless.

Reply:
Alright, the "avoiding `bytes` conversion fixes both problems" is the part I didn't realize. :+1: